src/fides/api/alembic/migrations/versions/fd52d5f08c17_system_dictionary_data_migration.py

```diff
@@ -0,0 +1,374 @@
+"""system dictionary data migration
+
+Revision ID: fd52d5f08c17
+Revises: 39397820a0d5
+Create Date: 2023-08-04 14:14:32.421414
+
+"""
+import json
+from typing import Any, Dict, List, Optional, Tuple
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+from sqlalchemy import text
+from sqlalchemy.engine import Connection
+from sqlalchemy.sql.elements import TextClause
+
+from fides.config import CONFIG
+
+revision = "fd52d5f08c17"
+down_revision = "39397820a0d5"
+branch_labels = None
+depends_on = None
+
+JOINT_CONTROLLER_INFO = "joint_controller_info"
+DOES_INTERNATIONAL_TRANSFERS = "does_international_transfers"
+RESPONSIBILITY = "responsibility"
+REQUIRES_DATA_PROTECTION_ASSESSMENTS = "requires_data_protection_assessments"
+DPA_LOCATION = "dpa_location"
+DPA_PROGRESS = "dpa_progress"
+
+LEGAL_BASIS_FOR_PROCESSING = "legal_basis_for_processing"
+IMPACT_ASSESSMENT_LOCATION = "impact_assessment_location"
+PROCESSES_SPECIAL_CATEGORY_DATA = "processes_special_category_data"
+SPECIAL_CATEGORY_LEGAL_BASIS = "special_category_legal_basis"
+DATA_SHARED_WITH_THIRD_PARTIES = "data_shared_with_third_parties"
+THIRD_PARTIES = "third_parties"
+RETENTION_PERIOD = "retention_period"
+
+
+def _system_transform_joint_controller_info(
+    joint_controller_decrypted: Optional[Dict],
+    datasets_joint_controller_decrypted: List[Optional[Dict]],
+) -> Optional[str]:
+    """Migrating systems.joint_controller and datasets.joint_controller encrypted json string fields to
+    System.joint_controller_info which is just a string.
+
+    Prioritizing system joint_controller info if it exists, or taking the first non-null entry from datasets
+    otherwise.
+    """
+    new_joint_controller_info: Optional[str] = None
+    # Prioritize system data over dataset data
+    joint_controller_data = (
+        joint_controller_decrypted
+        or next(  # Multiple datasets can be related to a system, so we use the first non null one if applicable
+            (
+                item
+                for item in datasets_joint_controller_decrypted or []
+                if item is not None
+            ),
+            None,
+        )
+    )
+    if joint_controller_data and isinstance(joint_controller_data, dict):
+        separated_fields = [
+            joint_controller_data.get("name"),
+            joint_controller_data.get("address"),
+            joint_controller_data.get("email"),
+            joint_controller_data.get("phone"),
+        ]
+
+        new_joint_controller_info = "; ".join(
+            [elem for elem in separated_fields if elem and elem != "N/A"]
+        )
```

Comment:
alternatively could add this:

```suggestion
        ) or None
```

![image](https://github.com/ethyca/fides/assets/5105354/8c73342d-67d0-4746-9f20-60d934f94103)

Reply:
Yes that is a good alternative, I'll use that instead!